File: MapGUI3/src/Model/Statements/PrintStatement.java
package Model.Statements;

import Model.Heap;
import Model.Output;
import Model.ProgramState;
import Model.expressions.Expression;

public class PrintStatement implements IStatement {
    private Expression expr;

    public PrintStatement(Expression expr) {
        this.expr = expr;
    }

    public ProgramState execute(ProgramState ps) {
        Output<Integer> output = ps.getOutput();
        Heap ph=ps.getHeap();
        int result = expr.evaluate(ps.getSymTable(),ph);
        output.add(result);
        return ps;
    }

    public String toString() {
        return "Print(" + expr.toString() + ")";
    }

}

File: MapA7/MapA7/Model/Statements/PrintStatement.cs
using MapA7.Model.Expressions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Model.Statements
{
    public class PrintStatement:IStatement
    {
        private IExpression expression;

        public PrintStatement(IExpression expression) { this.expression = expression; }

        public ProgramState Execute(ProgramState ps)
        {
            Output output = ps.GetOutput();
            SymbolTable symbolTable = ps.GetSymbolTable();
            int result = expression.Evaluate(symbolTable);
            output.Add(result.ToString());
            return ps;
        }

        public String Tostring() { return "Print(" + expression.Tostring() + ')'; }
    }
}
